Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -484,6 +484,39 @@ void webkitWebViewBaseToplevelWindowIsActiveChanged(WebKitWebViewBase* webViewBa
     priv->pageProxy->activityStateDidChange(ActivityState::WindowIsActive);
 }
 
+static void webkitWebViewBaseUpdateVisibility(WebKitWebViewBase* webViewBase)
+{
+    WebKitWebViewBasePrivate* priv = webViewBase->priv;
+    bool isVisible = [&]() {
+        if (!gtk_widget_get_mapped(GTK_WIDGET(webViewBase)))
+            return false;
+
+        if (!priv->toplevelOnScreenWindow->isInMonitor())
+            return false;
+
+        if (priv->toplevelOnScreenWindow->isMinimized())
+            return false;
+
+        if (priv->toplevelOnScreenWindow->isSuspended())
+            return false;
+
+        return true;
+    }();
```

Comment:
Any reason for the lambda instead of calculating the `isVisible` directly? Maybe it's only a matter of taster, but I find the following more readable:

```suggestion
    const bool isVisible = gtk_widget_get_mapped(GTK_WIDGET(webViewBase) &&
        priv->toplevelOnScreenWindow->isInMonitor() &&
        !priv->toplevelOnScreenWindow->isMinimized() &&
        !priv->toplevelOnScreenWindow->isSuspended();
```

Reply:
In my initial patch the lambda was more complex, and then I simplified everything and kept it. I agree in its current state your proposal looks much better. I'll update the patch. Thanks!